sources/api/netdog/src/wicked.rs

```diff
@@ -0,0 +1,280 @@
+    mode: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    link_detection: Option<String>,
+    // `serde_xml_rs` has a known issue with serializing nested structures, where it will insert
+    // additional tag with the structure name.  It has since been fixed but not released officially
+    // yet.  This struct member works around that issue.
+    // https://github.com/RReverser/serde-xml-rs/issues/126
+    // The workaround:
+    // https://stackoverflow.com/questions/70124048/how-to-create-xml-from-struct-in-rust
+    #[serde(flatten, skip)]
+    _f: (),
+}
+
+// These options are only used in AWS variants
+#[derive(Default, Debug, Serialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+struct WickedIpv4 {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    arp_verify: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    arp_notify: Option<bool>,
+    #[serde(flatten, skip)]
+    _f: (),
+}
+
+#[derive(Default, Debug, Serialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct WickedDhcp4 {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    enabled: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    route_priority: Option<u32>,
+    #[serde(flatten, skip)]
+    _f: (),
+}
+
+#[derive(Default, Debug, Serialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct WickedDhcp6 {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    enabled: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    defer_timeout: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    // Technically "flags" is an enum, but considering that we don't expose this to the user,
+    // provide a default for it, and the flag itself is a self-closing tag, we opt to keep it a
+    // String instead of hacking around serde/XML's "interesting" handling of enums/tags
+    flags: Option<String>,
+    #[serde(flatten, skip)]
+    _f: (),
+}
+
+// We assume that all configured interfaces are wanted at boot and will require a link to
+// be considered configured
+impl Default for WickedControl {
+    fn default() -> Self {
+        WickedControl {
+            mode: Some("boot".to_string()),
+            link_detection: Some("<require-link />".to_string()),
+            _f: (),
+        }
+    }
+}
+
+impl From<Dhcp4> for WickedDhcp4 {
+    fn from(dhcp4: Dhcp4) -> Self {
+        match dhcp4 {
+            Dhcp4::DhcpEnabled(b) => WickedDhcp4 {
+                enabled: Some(b),
+                _f: (),
+                ..Default::default()
+            },
+            Dhcp4::WithOptions(o) => WickedDhcp4 {
+                enabled: o.enabled,
+                route_priority: o.route_metric,
+                _f: (),
+            },
+        }
+    }
+}
+
+// The default configuration for AWS based variants
+#[cfg(any(bottlerocket_platform = "aws", bottlerocket_platform = "vmware"))]
```

Comment:
Nit: the comment and the code aren't consistent. Should the `vmware` flag be used below, with the `metal` platform?

Reply:
Can you reduce the conditionally-compiled code even further to a set of constants?

```rust
#[cfg(any(bottlerocket_platform = "aws", bottlerocket_platform = "vmware"))]
const DEFAULT_WICKED_IPV4: Option<WickedIpv4> = Some(WickedIpv4{ /* ... */ });

#[cfg(bottlerocket_platform = "metal")]
const DEFAULT_WICKED_IPV4: Option<WickedIpv4> = Some(WickedIpv4{ /* ... */ });

impl Default for WickedInterface {
    fn default() -> Self {
WickedInterface {
            name: DEFAULT_INTERFACE.to_string(),
            control: WickedControl::default(),
            ipv4: DEFAULT_WICKED_IPV4,
           /* etc */
        }
    }
}
```

You may also be able to emit a useful error message in the case that someone is trying to define a new variant. (This is pseudo code for the cfg part...)

```rust
#[cfg(none_of(bottlerocket_platform = "aws", bottlerocket_platform = "vmware", bottlerocket_platform = "metal"))]
compile_error!("Unexpected bottlerocket_platform. Default Wicked settings must be defined.");
```